src/main/webapp/app/overview/course-exercises/course-exercise-row.component.ts

```diff
@@ -51,6 +51,7 @@ export class CourseExerciseRowComponent implements OnInit, OnDestroy, OnChanges
         private exerciseService: ExerciseService,
         private httpClient: HttpClient,
         private participationWebsocketService: ParticipationWebsocketService,
+        private profileService: ProfileService,
```

Comment:
Why do you need the profileService here?

Reply:
I'm still looking if there is a better place, but right now it seems like I need this service (and the call to `setBuildPlanUrlForProgrammingParticipations` twice). Once when getting participation updates (after pressing start or resume) and once for the initial page load (if the exercise is already started). It seems like these two are completely different paths, but again I'm still looking into this.